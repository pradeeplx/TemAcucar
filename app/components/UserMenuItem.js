import React from 'react'
import { View, TouchableOpacity } from 'react-native'
import { fontFactor } from "../helpers"
import Colors from "../Colors"
import Sentence from "./Sentence"
import Icon from "./Icon"

export default UserMenuItem = ({ onPress, style, icon, iconSet, iconStyle, children, type }) => (
  <TouchableOpacity onPress={onPress} style={[{
    alignSelf: 'stretch',
    padding: 10,
    paddingBottom: 0,
    flexDirection: 'row',
    alignItems: 'center'
  }, style]}>
    <View style={{
      alignItems: 'center',
      width: 24 * fontFactor(),
    }}>
      <Icon name={icon} set={iconSet} style={[{
        fontSize: 24 * fontFactor(),
        color: (type == "light" ? Colors.white : Colors.white),
      }, iconStyle]} />
    </View>
    <Sentence style={{
      marginLeft: 8,
      fontSize: 16 * fontFactor(),
      color: (type == "light" ? Colors.white : Colors.white),
    }}>
      {children}
    </Sentence>
  </TouchableOpacity>
)
