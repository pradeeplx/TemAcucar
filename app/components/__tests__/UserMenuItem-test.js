import 'react-native'
import React from 'react'
import UserMenuItem from '../UserMenuItem'
import renderer from 'react-test-renderer'

const props = {
  style: { top: 0 },
  icon: 'ios-add',
  iconStyle: { top: 1 },
  children: 'FooBar',
}

it('renders correctly', () => {
  const tree = renderer.create(
    <UserMenuItem {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
