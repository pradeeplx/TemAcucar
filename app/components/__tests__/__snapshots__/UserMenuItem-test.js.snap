exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "flexDirection": "row",
      "opacity": 1,
      "padding": 10,
      "paddingBottom": 0,
      "top": 0,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "width": 36,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Ionicons",
            "fontSize": 27,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 36,
            },
            Object {
              "top": 1,
            },
          ],
        ]
      }>
      ï„‚
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "marginLeft": 8,
        },
      ]
    }>
    FooBar
  </Text>
</View>
`;
